### ISSUE TITLE
[Detector Support]: OpenVINO with Intel iGPU not working

### ISSUE THREAD

**Issue by esand** [blakeblackshear/frigate#9641]
### Describe the problem you are having

I previously had Frigate 0.12 running with my detector set to openvino using the standard config from the docs. Everything worked fine; Inference was very low, CPU usage was decent and object detection worked.

Upgrading to Frigate 0.13.1 (stable-tensorrt) with the same setup and it fails to start completely. It errors out trying to build some tensorrt models or whatever which was never an issue in Frigate 0.12. I don't have an NVIDIA GPU, just the Intel CPU with UHD graphics and so I figure that could be why the models aren't building successfully, but all that is way beyond me.

I tried following some commands as per the docs to try and install yolox-tiny, but I had no idea what I was doing and in the end it was a hot mess so I gave up (literally, no clue what I was doing).

Providing the relevant config piece. I use the tensorrt container because of my need for the vaapi stuff for hardware accel - it doesn't work (missing drivers and such) in the base stable container image. I tried playing with the YOLO_MODELS env variable (blank, a few other models) but they all seemed to fail.

### Version

0.13.1

### Frigate config file

```yaml
detectors:
  ov:
    type: openvino
    device: AUTO
    model:
      path: /openvino-model/ssdlite_mobilenet_v2.xml

model:
  width: 300
  height: 300
  input_tensor: nhwc
  input_pixel_format: bgr
  labelmap_path: /openvino-model/coco_91cl_bkgr.txt

ffmpeg:
  hwaccel_args: preset-vaapi
  output_args:
    record: preset-record-generic-audio-copy
```

### docker-compose file or Docker CLI command

```yaml
version: "3.8"
services:
  frigate:
    container_name: frigate
    privileged: false
    restart: unless-stopped
    image: "ghcr.io/blakeblackshear/frigate:stable-tensorrt"
    shm_size: "128mb" # ~12mb per 720p detect stream
    ports:
      - "5000:5000"
      - "8554:8554" # RTSP feeds
      - "8555:8555/tcp" # WebRTC over tcp
      - "8555:8555/udp" # WebRTC over udp
    devices:
      - /dev/dri/card0
      - /dev/dri/renderD128
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /home/docker/frigate/config:/config
      # /home/docker/frigate/config.yml:/config/config.yml:ro
      # /home/docker/frigate/go2rtc:/config/go2rtc:ro
      - /mnt/storage/frigate/media:/media/frigate
      # /mnt/storage/frigate/db:/db
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    cap_add:
      - CAP_PERFMON
    group_add:
      - "44" # video
      - "105" # render
    environment:
      FRIGATE_RTSP_PASSWORD: "password"
      YOLO_MODELS: "yolov7-320"
    networks:
      ipvlan:
        ipv4_address: 192.168.x.y
      local:
        ipv4_address: 172.18.x.y
networks:
   ipvlan:
     name: cameras
     external: true
   local:
     name: local
     external: true
```

### Relevant log output

```shell
Creating yolov7-320.cfg and yolov7-320.weights
Done.
2024-02-03 18:56:24.661951805  [INFO] Starting go2rtc healthcheck service...
Generating yolov7-320.trt. This may take a few minutes.
Traceback (most recent call last):
  File "/usr/local/src/tensorrt_demos/yolo/onnx_to_tensorrt.py", line 214, in <module>
    main()
  File "/usr/local/src/tensorrt_demos/yolo/onnx_to_tensorrt.py", line 202, in main
    engine = build_engine(
  File "/usr/local/src/tensorrt_demos/yolo/onnx_to_tensorrt.py", line 112, in build_engine
    with trt.Builder(TRT_LOGGER) as builder, builder.create_network(*EXPLICIT_BATCH) as network, trt.OnnxParser(network, TRT_LOGGER) as parser:
TypeError: pybind11::init(): factory function returned nullptr
[02/03/2024-18:56:33] [TRT] [W] Unable to determine GPU memory usage
[02/03/2024-18:56:33] [TRT] [W] Unable to determine GPU memory usage
[02/03/2024-18:56:33] [TRT] [W] CUDA initialization failure with error: 35. Please check your CUDA installation:  http://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html
Loading the ONNX file...
Available tensorrt models:
ls: cannot access '*.trt': No such file or directory
s6-rc: warning: unable to start service trt-model-prepare: command exited 2
```

### Operating system

Proxmox

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Any other information that may be helpful

go2rtc works, but Frigate fails to start up because it can't compile the tensorrt stuff. Do I even need that for Intel built-in graphics?
As an FYI, in 0.12 when I had this working, inference was at ~18ms compared to 210+ for just plain-jane CPU which is why I really want this config to still work. I have confirmed that openvino should work for my GPU as per Intel docs, so I know it's capable.

**Comment by NickM-27** [comment]
> I don't have an NVIDIA GPU, just the Intel CPU with UHD graphics and so I figure that could be why the models aren't building successfully, but all that is way beyond me.

Why are you running the tensorrt image if you don't have an nvidia gpu? This has nothing to do with hwaccel and only has to do with object detection via nvidia gpu

**Comment by esand** [comment]
> Why are you running the tensorrt image if you don't have an nvidia gpu? This has nothing to do with hwaccel and only has to do with object detection via nvidia gpu
The regular image doesn't have intel-media-va-driver-non-free included and when I was trying to get ffmpeg hardware acceleration to work in previous Frigate versions, I needed the tensorrt container to get it all to work.

If I try my config on the standard container, I get an error about specifying model settings other than just the path anyways... so I assumed I still needed the tensorrt container to use openvino as well.

If you're telling me that I can define my detector to be openvino as per docs, and I can use hwaccel settings that need the intel-media-va-driver-non-free package and use the standard container, I'll try it... but I assume I'd need to do *something* to either my config or the container (or both) to get them to work and I'm not sure what that is (many sleepless nights fighting with configs to get hardware accel to work properly).

**Comment by esand** [comment]
Ok - so I seem to have misspoke there. I had issues getting my config running in previous Frigate containers, however I just tried the standard container with no database and got it working. Not sure what I did slightly different this time around - perhaps it was just the database issue I just encountered which caused a boot loop.

The "database issue" is that it seems specifying
`database:
  path: /db/frigate.db`
in your config file doesn't seem to work - Frigate complains it can't find your database file... but separate issue.
